src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/FailoverConnectionPlugin.java

```diff
@@ -1334,14 +1314,20 @@ private void validateHostPatternSetting(String hostPattern) throws SQLException
     }
   }
 
-  private Map<String, String> getInitialConnectionProps(final PropertySet propertySet) {
+  private Map<String, String> getInitialConnectionProps(final PropertySet propertySet, HostInfo currentHost) {
     final Map<String, String> initialConnectionProperties = new HashMap<>();
     final Properties originalProperties = propertySet.exposeAsProperties();
     originalProperties.stringPropertyNames()
         .stream()
         .filter(x -> this.propertySet.getProperty(x).isExplicitlySet())
         .forEach(x -> initialConnectionProperties.put(x, originalProperties.getProperty(x)));
 
+    initialConnectionProperties.putAll(currentHost.getHostProperties());
+    initialConnectionProperties.put(PropertyKey.USER.getKeyName(), currentHost.getUser());
+    initialConnectionProperties.put(PropertyKey.PASSWORD.getKeyName(), currentHost.getPassword());
```

Comment:
Would we ever run into the situation where the user/password are set in the current host properties but getUser/getPassword return null?

Reply:
I don't think so, the driver checks host properties for the user/password before this point and will use the default user/password if nothing is there.